### PR TITLE
【doc】add the comments of device-ipc completion

### PR DESCRIPTION
**Pull request by chenyingchun0312**
## 拉取/合并请求描述：(PR description)

[
添加completion注释
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [x] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [x] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [x] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [x] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [x] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [x] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [x] 本拉取/合并符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

File: components/drivers/src/completion.c [hunk 1]
```diff
@@ -15,6 +15,11 @@
 #define RT_COMPLETED    1
 #define RT_UNCOMPLETED  0
 
+/**
+ * @brief initialize the completion object
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:19]: 请检查：句子开头是大写，结束有句号

File: components/drivers/src/completion.c [hunk 2]
```diff
@@ -27,6 +32,19 @@ void rt_completion_init(struct rt_completion *completion)
 }
 RTM_EXPORT(rt_completion_init);
 
+/**
+ * @brief waitting for the completion done
+ *        NOTE: We should not use this api when
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:37]: fragment，when什么？

File: components/drivers/src/completion.c [hunk 3]
```diff
@@ -27,6 +32,19 @@ void rt_completion_init(struct rt_completion *completion)
...
+ *
+ * @param completion the point of completion object
+ * @param timeout    is a timeout period (unit: an OS tick). If the completion is unavailable, the thread will wait for
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:40]: OS ticks

File: components/drivers/src/completion.c [hunk 4]
```diff
@@ -27,6 +32,19 @@ void rt_completion_init(struct rt_completion *completion)
...
+ *                   the completion up to the amount of time specified by the argument.
+ *                   NOTE: Generally, we use the macro RT_WAITING_FOREVER to set this parameter, which means that when the
+ *                   completion is unavailable, the thread will be waitting forever.
+ *
+ * @return Return the operation status. ONLY When the return value is RT_EOK, the operation is successful.
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:45]: When 小写

File: components/drivers/src/completion.c [hunk 5]
```diff
@@ -27,6 +32,19 @@ void rt_completion_init(struct rt_completion *completion)
...
+ *         If the return value is any other values, it means that the completion wait failed.
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:46]: it means that waiting for the completion failed.

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: components/drivers/src/completion.c (hunk 1 above)
Comment (chenyingchun0312) [reply components/drivers/src/completion.c:19]: 好的，收到，请问下，哪里有比较规范的注释要求，说明文档？

File: components/drivers/src/completion.c (hunk 3 above)
Comment (chenyingchun0312) [reply components/drivers/src/completion.c:40]: 谢谢满老师这么细心的审核

File: components/drivers/src/completion.c [hunk 6]
```diff
@@ -27,6 +33,22 @@ void rt_completion_init(struct rt_completion *completion)
 }
 RTM_EXPORT(rt_completion_init);
 
+/**
+ * @brief This function will wait for a completion, if the completion is unavailable, the thread shall wait for
+ *        the completion up to a specified time.
+ *
+ * @param completion is a pointer to a completion object.
+ *
+ * @param timeout is a timeout period (unit: an OS ticks). If the completion is unavailable, the thread will wait for
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:42]: 不不，这个是OS ticks 没有an了

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: components/drivers/src/completion.c [hunk 7]
```diff
@@ -27,6 +33,22 @@ void rt_completion_init(struct rt_completion *completion)
...
+ *                the completion done up to the amount of time specified by the argument.
+ *                NOTE: Generally, we use the macro RT_WAITING_FOREVER to set this parameter, which means that when the
+ *                completion is unavailable, the thread will be waitting forever.
+ *
+ * @return Return the operation status. ONLY When the return value is RT_EOK, the operation is successful.
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:47]: when小写没改过来

**Review by mysterywolf — CHANGES_REQUESTED** [review]

File: components/drivers/src/completion.c [hunk 8]
```diff
@@ -27,6 +33,22 @@ void rt_completion_init(struct rt_completion *completion)
...
+ *         If the return value is any other values, it means that the completion wait failed.
+ *
+ * @warning This function can ONLY be called in the thread context. It MUST NOT BE called in interrupt context.
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:50]: It MUST NOT be，be不用大写

File: components/drivers/src/completion.c [hunk 9]
```diff
@@ -94,6 +116,11 @@ rt_err_t rt_completion_wait(struct rt_completion *completion,
 }
 RTM_EXPORT(rt_completion_wait);
 
+/**
+ * @brief This function indicate a completion has done.
```
Comment (mysterywolf) [inline components/drivers/src/completion.c:120]: indicates

**Review by mysterywolf — CHANGES_REQUESTED** [review]
其他没啥问题了

**Comment by chenyingchun0312** [comment]
> 其他没啥问题了

好的，我修改下，再提交下

**Review by mysterywolf — APPROVED** [review]

**Pull request merged**